packages/nuxt/src/pages/runtime/page.ts

```diff
@@ -14,6 +14,8 @@ import { _wrapIf } from '#app/components/utils'
 import { LayoutMetaSymbol, PageRouteSymbol } from '#app/components/injections'
 // @ts-expect-error virtual file
 import { appKeepalive as defaultKeepaliveConfig, appPageTransition as defaultPageTransition } from '#build/nuxt.config.mjs'
+// @ts-expect-error virtual file
+import { ClientOnly } from '#build/pages-wrapper.mjs'
```

Comment:
So we bundle `createClientOnly` even if the page is not client only in some cases ?

Reply:
It should bundle `ClientOnly` component only if at least one of the pages are client only

Reply:
yeah... can't we instead wrap the page component competly with `createClientOnly` ?

Because we'd bundle `ClientOnly` even if we're going to a page that is not client only

Reply:
Daniel also suggested that 😅. It did work, but the main problem was that the Vue showed warnings with page transitions enabled. Client only component isn't working well under transitions :(

Reply:
I see, maybe can we wrap with a div so vue can correctly apply transitions ?

Reply:
Tried doing that and now even just wrapping page component into `createClientOnly` seems not working 😅

Reply:
I'm currently testing island pages by defining it within the `route.mjs` with a dynamic import and a `.then`. I'm pretty sure we can do the same thing for `createClientOnly`